GVFS/GVFS.FunctionalTests/Tools/GVFSHelpers.cs

```diff
@@ -102,16 +102,18 @@ public static string ReadAllTextFromWriteLockedFile(string filename)
             }
         }
 
-        public static void ModifiedPathsContentsShouldEqual(FileSystemRunner fileSystem, string dotGVFSRoot, string contents)
+        public static void ModifiedPathsContentsShouldEqual(GVFSFunctionalTestEnlistment enlistment, FileSystemRunner fileSystem, string contents)
         {
-            string modifiedPathsDatabase = Path.Combine(dotGVFSRoot, TestConstants.Databases.ModifiedPaths);
+            enlistment.WaitForBackgroundOperations().ShouldBeTrue("Background operations failed to complete.");

```

Comment:
It looks like every single call to `WaitForBackgroundOperations` throughout our test code always has to append the `ShouldBeTrue("same error message")`. I would move that check inside the `WaitForBackgroundOperations` method and get rid of all of that duplication.